lib/pdf/elements/list.rb

```diff
@@ -5,12 +5,17 @@ def self.render(pdf, element, indent: 0)
     index = 0
     pdf.indent indent do
       element.css("> li").each do |li|
-        inner_list = li.children.css("ul,ol")
+        text = li.children
```

Comment:
You are changing the order of the children here, you first render all other nodes, than later render the `ul` or `ol`.

The problem lies with you needing to "print" every child recursively.